commons-numbers-complex-arrays/src/test/java/org/apache/commons/numbers/complex/arrays/ComplexListTest.java

```diff
@@ -130,14 +131,12 @@ void testSetAddAndAddAllNullPointerException() {
         Assertions.assertThrows(NullPointerException.class, () -> list1.add(0, null));
         Assertions.assertThrows(NullPointerException.class, () -> list1.set(1, null));
 
+        ComplexList copy = ComplexList.interleaved();
+        copy.addAll(list1);
         List<Complex> list2 = generateList(3);
-        for (int i = 0; i < list2.size(); i++) {
-            Complex val = list2.get(i);
-            if (val != null) {
-                list2.set(i, null);
-            }
-        }
+        list2.set(1, null);
         Assertions.assertThrows(NullPointerException.class, () -> list1.addAll(list2));
+        Assertions.assertEquals(copy.size(), list1.size());
```

Comment:
You should also check `addAll(int, Collection)`. The implementation, if done incorrectly, could move all elements to make space, then copy the new elements in and throw a NPE. This will leave the list in a broken state.

It may be better to check the contents of the list are the same using `Assertions.assertEquals(copy, list1);`